app/src/main/java/com/conupods/OutdoorMaps/View/MainActivity.java

```diff
@@ -19,34 +21,27 @@
 
     private static final String TAG = "MainActivity";
     private static final int ERROOR_DIALOG_REQUEST = 9001;
-    Button MapActivityButton;
+    Handler launchMaps;
 
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        if(isGoogleAPIServiceAvailable())
-            init();
-
-
-    }
-
-
-    private void init(){
-        MapActivityButton = (Button) findViewById(R.id.MapActivityButton);
-        MapActivityButton.setOnClickListener(new View.OnClickListener() {
-
-            @Override
-            public void onClick(View view){
-                startActivity(new Intent(MainActivity.this, MapsActivity.class));
-
-            }
-        });
+        if(isGoogleAPIServiceAvailable()) {
+            launchMaps=new Handler();
+            launchMaps.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    Intent intent=new Intent(MainActivity.this,MapsActivity.class);
+                    startActivity(intent);
+                    finish();
+                }
+            },2000);
+        }
     }
```

Comment:
shouldn't you keep the loading screen ongoing until googleAPIServiceIsAvailable? AKA shouldnt  that if statement should be inside run() method. Also, shouldn't you use lambda functions as they are much cleaner

Reply:
Changes are committed. I couldnt checkout v2 itself to confirm, but works well on my other branch.